fastestimator/network/lrschedule.py

```diff
@@ -0,0 +1,56 @@
+import math
+
+import numpy as np
+
+
+class LRSchedule:
+    def __init__(self, schedule_mode):
+        self.schedule_mode = schedule_mode
+        self.total_epochs = None  #will be filled by runtime
+        self.total_steps = None  #will be filled by runtime
+        self.initial_lr = None  #will be filled by runtime
+
+    def schedule_fn(self, current_step_or_epoch, lr):
+        #do something to change lr
+        return lr
+
+
+class CyclicLRSchedule(LRSchedule):
```

Comment:
do you think it would be worth implementing the math functions here as @tf.function since they might get invoked at every step?

Reply:
i don't think so, the computation is just a scalar, cpu is better for it.   

gpu is good for large data when data parallelism can be used.